db/users.go

```diff
@@ -0,0 +1,30 @@
+package db
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+// GetUserID obtains the internal user ID  for the given username and user domain from the DE database.
+func GetUserID(tx *sql.Tx, username, userDomain string) (string, error) {
+	query := "SELECT id FROM users WHERE username = $1"
+	qualifiedUsername := fmt.Sprintf("%s@%s", username, userDomain)
+
+	// Query the database.
+	rows, err := tx.Query(query, qualifiedUsername)
```

Comment:
Since this is selecting one row, it might be cleaner to use [QueryRow](https://golang.org/pkg/database/sql/#Tx.QueryRow) -- or, if we have a `context.Context` when using this (not sure, but I wouldn't be shocked if echo is smart enough to make one per request), `QueryRowContext` which would know to cancel the query/not perform it if the request is already dead. `Query` also has a `QueryContext` equivalent that might be worth looking into using.

Reply:
I didn't see an instance of `context.Context` per request in the Echo source. An instance of `context.Context` is used in the function to shut down the server, but that wouldn't apply to a single request.

I've used `QueryRow` before, and I wouldn't object to using it here. I don't have much of an opinion about which is cleaner.